sdk/scheduler/src/main/java/com/mesosphere/sdk/api/PodResource.java

```diff
@@ -299,14 +288,23 @@ private Response restartPod(String podInstanceName, RecoveryType recoveryType) {
             return ResponseUtils.elementNotFoundResponse();
         }
```

Comment:
Is there any problem with invoking the listener even if taskKiller is null? Also probably can't hurt to move the log call above this

Reply:
There probably isn't a problem as they'll be detected as needing recovery later and killed anyway.  But for consistency / ease of understanding, I think you're right.  We'll do both or neither.